### ISSUE TITLE
Supporting HPA autoscaling/v2

### ISSUE THREAD

**Issue by pseudonator** [solo-io/gloo#7634]
### Version

1.12.x (latest stable)

### Is your feature request related to a problem? Please describe.

HPA has been graduated to GA and we should use autoscaling/v2 for all the HPAs that we create on Kubernetes v1.23+. This has been widely [blogged](https://kubernetes.io/blog/2021/12/07/kubernetes-1-23-release-announcement/#horizontalpodautoscaler-v2-graduates-to-ga) as well.

This means that when Gloo is deployed with its current HPA  on Kubernetes v1.23+, Kubernetes API will auto update either `autoscaling/v1` or `autoscaling/v2beta2` automatically. Downside to this is there currently isn't a good set of options to configure HPA for `autoscaling/v2` version.

### Describe the solution you'd like

[Gloo HPA configuration](https://github.com/solo-io/gloo/blob/v1.13.2/install/helm/gloo/templates/8-gateway-proxy-horizontal-pod-autoscaler.yaml) should include `autoscaling/v2`. 

### Additional Context

Support matrix already includes Kubernetes 1.23 & 1.24 as supported.

**Comment by kevin-shelaga** [comment]
I could probably add this in fairly quickly, but what fields are missing right now? Metrics and behaviour are both there and available to be defined from what I can see @pseudonator 

https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2/#HorizontalPodAutoscalerSpec

**Comment by pseudonator** [comment]
@kevin-shelaga yup both configs are supported but adding `autoscaling/v2` should be suffic at this point. If you don't have any bandwidth let me know I'm happy to raise a PR for this.

In addition IMO we should add a page in the docs about this so people are aware if they are running in 1.23+ to specify `autoscaling/v2` instead.

**Comment by kevin-shelaga** [comment]
@pseudonator my mistake i misread the template, i see it currently checks for v2beta for those fields. Makes sense now.
